### DISCUSSION TITLE
Low quality transcoding results despite high bitrate and slow preset configured.

### DISCUSSION THREAD

**Discussion by Baa14453** [OvenMediaLabs/OvenMediaEngine#1598] [Help] [answered]
Hi, I'm having an issue where no matter what settings I configure for OME transcoding the quality comes out really poor.
I am using the OME Docker Container within Kubernetes.
This is my transcoding config:
```
<OutputProfiles>
<OutputProfile>
	<Name>legit</Name>
	<OutputStreamName>${OriginStreamName}</OutputStreamName>
	<!-- LLHLS -->
	<Playlist>
		<Name>LLHLS-Stable</Name>
		<FileName>llhls</FileName>
		<Rendition>
			<Name>A-1080p PREMIUM</Name>
			<Video>1080p+</Video>
			<Audio>512k-5.1-AAC</Audio>
		</Rendition>
	</Playlist>

	<!-- WebRTC playlist-->
	<Playlist>
		<Name>Webrtc-Test</Name>
		<FileName>webrtc</FileName>
		<Options>
			<WebRtcAutoAbr>false</WebRtcAutoAbr>
		</Options>
		<Rendition>
			<Name>A-1080p PREMIUM</Name>
			<Video>1080p+</Video>
			<Audio>512k-5.1-Opus</Audio>
		</Rendition>
	</Playlist>

	<Encodes>
		<!-- 1080p+ -->
		<Video>
			<Name>1080p+</Name>
			<Codec>h264_openh264</Codec>
			<Width>1920</Width>
			<Height>1080</Height>
			<Bitrate>12000000</Bitrate>
			<Framerate>30.0</Framerate>
			<KeyFrameInterval>30</KeyFrameInterval>
			<BFrames>0</BFrames>
			<Profile>baseline</Profile>
			<Preset>slower</Preset>
		</Video>
		<Audio>
			<Name>512k-5.1-AAC</Name>
			<Codec>aac</Codec>
			<Bitrate>512000</Bitrate>
			<Samplerate>48000</Samplerate>
			<Channel>6</Channel>
		</Audio>
		<Audio>
			<Name>512k-5.1-Opus</Name>
			<Codec>opus</Codec>
			<Bitrate>512000</Bitrate>
			<Samplerate>48000</Samplerate>
			<Channel>6</Channel>
		</Audio>
	</Encodes>
</OutputProfile>
</OutputProfiles>
```
This is the original footage:

https://github.com/AirenSoft/OvenMediaEngine/assets/9057997/78bad022-cb72-4f10-9eb8-26b533625026

This is what it looks like when I record direct from OBS (to ensure OBS wasn't compressing the video loads):

https://github.com/AirenSoft/OvenMediaEngine/assets/9057997/40df4766-0940-4378-ae7c-c4a91a8fa972

This is what it looks like when I record the LLHLS stream after OME transcoding. 
The quality has noticeably diminished with considerably more compression artefacts than I would expect for 12,000Kbps footage with the Slower preset.

https://github.com/AirenSoft/OvenMediaEngine/assets/9057997/106fc825-69a1-44a3-b325-07b4aad741b5

In addition, the bitrate recorded in the file is only about 8,000Kbps and not the 12,000 I would expect.
I tried a number of different settings to help, including increasing the bitrate up to 30,000Kbps, and multiple presets such as `fast` but the quality remained poor throughout them all.

**Comment by getroot** [comment, marked as answer]
This appears to be the difference between openh264 and x264 used in OBS. Using a hardware encoder like NVIDIA will most likely give you the results you want.

**Reply by Baa14453** [reply]
It's a rented server so I don't have access to modify the hardware, I just have the XEON CPU to work with. Is there any way I can use x264?

**Reply by Baa14453** [reply]
Until x264 can be re-implemented I'm going to move to a different product, this is very disappointing as OME is better in every other way but the quality loss of openh264 is awful :(
Best of luck to you
